include/cuco/static_multimap.cuh

```diff
@@ -0,0 +1,1852 @@
+     */
+    template <typename KeyEqual>
+    __device__ insert_result cas_dependent_write(iterator current_slot,
+                                                 value_type const& insert_pair,
+                                                 KeyEqual key_equal) noexcept;
+
+    /**
+     * @brief Inserts the specified key/value pair into the map using vector loads.
+     *
+     * @tparam uses_vector_load Boolean flag indicating whether vector loads are used or not
+     * @tparam CG Cooperative Group type
+     * @tparam KeyEqual Binary callable type
+     *
+     * @param g The Cooperative Group that performs the insert
+     * @param insert_pair The pair to insert
+     * @param precomputed_hash Optional value which allows users to specify the precomputed hash
+     * value
+     * @param key_equal The binary callable used to compare two keys for
+     * equality
+     * @return void.
+     */
+    template <bool uses_vector_load, typename CG, typename KeyEqual = thrust::equal_to<key_type>>
+    __device__ std::enable_if_t<uses_vector_load, void> insert_impl(
+      CG g,
+      value_type const& insert_pair,
+      optional_hash_type precomputed_hash,
+      KeyEqual key_equal = KeyEqual{}) noexcept;
+
+    /**
+     * @brief Inserts the specified key/value pair into the map using scalar loads.
+     *
+     * @tparam uses_vector_load Boolean flag indicating whether vector loads are used or not
+     * @tparam CG Cooperative Group type
+     * @tparam KeyEqual Binary callable type
+     *
+     * @param g The Cooperative Group that performs the insert
+     * @param insert_pair The pair to insert
+     * @param precomputed_hash Optional value which allows users to specify the precomputed hash
+     * value
+     * @param key_equal The binary callable used to compare two keys for
+     * equality
+     * @return void.
+     */
+    template <bool uses_vector_load, typename CG, typename KeyEqual = thrust::equal_to<key_type>>
+    __device__ std::enable_if_t<not uses_vector_load, void> insert_impl(
+      CG g,
+      value_type const& insert_pair,
+      optional_hash_type precomputed_hash,
+      KeyEqual key_equal = KeyEqual{}) noexcept;
+
+   public:
+    /**
+     * @brief Inserts the specified key/value pair into the map.
+     *
+     * @tparam CG Cooperative Group type
+     * @tparam KeyEqual Binary callable type
+     *
+     * @param g The Cooperative Group that performs the insert
+     * @param insert_pair The pair to insert
+     * @param precomputed_hash Optional value which allows users to specify the precomputed hash
+     * value
+     * @param key_equal The binary callable used to compare two keys for
+     * equality
+     * @return void.
+     */
+    template <typename CG, typename KeyEqual = thrust::equal_to<key_type>>
+    __device__ void insert(CG g,
+                           value_type const& insert_pair,
+                           optional_hash_type precomputed_hash,
```

Comment:
Is this leftover? We should not need any precomputed_hash function arguments. That should all be able to be done via the passed in hash function.